services/victoriametrics/victoriametrics.go

```diff
@@ -84,7 +84,7 @@ func NewVictoriaMetrics(scrapeConfigPath string, db *reform.DB, baseURL string,
 		client:           new(http.Client),
 		baseConfigPath:   params.BaseConfigPath,
 		l:                logrus.WithField("component", "victoriametrics"),
-		sema:             make(chan struct{}, 1),
+		sema:             make(chan struct{}),
```

Comment:
Well, it was buffered :) And should be buffered

Reply:
But if it's buffered that batch protection for configuration update doesn't work. I'm able to trigger config reloading twice (one by one) and it will reloaded twice

Reply:
Reverted